src/main/java/graphql/execution/Execution.java

```diff
@@ -79,6 +81,12 @@ public Execution(ExecutionStrategy queryStrategy, ExecutionStrategy mutationStra
                 .document(document)
                 .operationDefinition(operationDefinition)
                 .build();
+
+
+        InstrumentationExecutionParameters parameters = new InstrumentationExecutionParameters(
+                executionInput, graphQLSchema, instrumentationState
+        );
+        executionContext = instrumentation.instrumentExecutionContext(executionContext, parameters);
         return executeOperation(executionContext, executionInput.getRoot(), executionContext.getOperationDefinition());
```

Comment:
Allows the execution context to be tweaked before execution via Instrumentation

Variables is one area I am thinking this would be used for